bin/src/app.rs

```diff
@@ -101,10 +101,21 @@ impl EthUIApp {
     }
 
     pub fn run(self) {
-        self.app.run(|_, event| {
-            if let tauri::RunEvent::ExitRequested { api, .. } = event {
+        self.app.run(|_, event| match event {
+            tauri::RunEvent::ExitRequested { api, .. } => {
                 api.prevent_exit();
             }
+
+            #[cfg(target_os = "macos")]
+            tauri::RunEvent::Reopen {
+                has_visible_windows,
+                ..
+            } => {
+                if !has_visible_windows {
+                    tokio::spawn(async { ethui_broadcast::main_window_show().await });
+                }
```

Comment:
this means if we click the icon while having a random dialog open, the main window won't show right? is this expected?

I would expect the intuitive behavior to be:
- open the main window, unless it's already open (but still open if other dialogs are open)
- if it's open but unfocused, focus it (not sure if this is doable)

but I'm not a mac user, so I'll defer to you

Reply:
can we have a dialog without the main window open? in what scenario would that appear?

Reply:
if you submit a tx via browser, while having the main window closed (and not being a plantext wallet in fast mode)

Reply:
this part still seems off. based on the previous discussion we'd want to trigger the logic regardless of if there are visible windows or not, right?